plugin-server/src/worker/ingestion/event-pipeline/1-emitToBufferStep.ts

```diff
@@ -21,8 +20,13 @@ export async function emitToBufferStep(
     const person = await runner.hub.db.fetchPerson(event.team_id, event.distinct_id)
 
     if (shouldBuffer(runner.hub, event, person, event.team_id)) {
-        const processEventAt = DateTime.now().plus({ seconds: runner.hub.BUFFER_CONVERSION_SECONDS })
-        await runner.hub.db.addEventToBuffer(event, processEventAt)
+        const processEventAt =
+            new Date(event.timestamp || Date.now()).valueOf() + runner.hub.BUFFER_CONVERSION_SECONDS * 1000
```

Comment:
### Code-level feedback:
This should have a test. If it's hard to test then extract this into a separate function and test that at least.

This code is so crufty I have no way of telling whether this is correct.

### Correctness-level feedback

Using event timestamp adds more questions than it answers. In a later step in the event pipeline we 'correct' the timestamp using a different method from this - should we be using that method consistently instead of adding one-off logic here?

Reply:
In the buffer 2.0 PR you asked me why I used the current time vs the event timestamp. I didn't really think about it and just used timestamp here.

However, based on this comment (great point!) I stopped to really think and decided that using `now` is just cleaner _and_ makes more sense.

Consider that if we've arrived here, it means we've decided the event needs to be buffered. As such, it makes no sense to use the timestamp as if it's backdated we'll just immediately process the event and the buffer will have no effect. 

Thus, I moved this to a simple `now + buffer_seconds`